tests-e2e/global-setup.ts

```diff
@@ -1,75 +1,145 @@
-import { ExecOptions } from "node:child_process"
-import * as http from "node:http"
 import * as path from "node:path"
 import execShCb from "exec-sh"
-import { TEST_SITES } from "./util"
 import handler from "serve-handler"
 import Table from "cli-table3"
 import * as color from "ansi-colors"
+
+import { TEST_SITES, LOCAL_OCW_PORT, siteUrl, TestSiteAlias } from "./util"
+import SimpleServer, { RedirectionRule } from "./util/SimpleServer"
 import { env } from "../env"
+import { hugo } from "../package_scripts/util"
 
 const execSh = execShCb.promise
 
+const STATIC_API_PORT = 4321
+
 /**
  * Resolve a path relative to package root.
  */
 const fromRoot = (...pathFromRoot: string[]) => {
   return path.join(__dirname, "../", ...pathFromRoot)
 }
-interface HugoOptions {
-  baseURL: string
-  themesDir: string
-  config: string
-  destination: string
-}
-const hugo = (hugoOptions: HugoOptions, execOptions: ExecOptions) => {
-  const flags = Object.entries(hugoOptions)
-    .map(([key, value]) => `--${key}=${value}`)
-    .join(" ")
-  return execSh(`yarn hugo ${flags} --verbose`, execOptions)
+const fromTmp = (...pathFromTmp: string[]) => {
+  return path.join(fromRoot("./test-sites/tmp"), ...pathFromTmp)
 }
 
-const buildSite = (name: string, configPath: string) => {
+const buildSite = (name: string, destInTmp: string, configPath: string) => {
   return hugo(
     {
       themesDir:   fromRoot("./"),
-      destination: fromRoot(`./test-sites/${name}/dist`),
-      baseURL:     `http://localhost:${env.WEBPACK_PORT}`,
-      config:      configPath
+      destination: fromTmp("dist", destInTmp),
+      cacheDir:    fromTmp("hugo_cache"),
```

Comment:
At one point when I was working on this, Hugo was caching nil responses. I think I had messed up one of the servers so, e.g., `localhost:4321/instructors/some-uuid/index.json` was returning nil incorrectly. Then Hugo cached that, so when I fixed the server, Hugo was still using the old response.

I went insane, then started deleting the cache, and then started putting the cache in the `tmp` directory (so I could delete it easily), and sanity returned.